tests/vxlan/test_vnet_route_leak.py

```diff
@@ -0,0 +1,184 @@
+
+    Args:
+        minigraph_facts: Minigraph information
+        duthost: DUT host object
+        vnet_config: Dictionary containing VNET configuration information
+        vnet_test_params: Dictionary containing VNET test parameters
+    """
+
+    logger.info("Backing up config_db.json")
+    duthost.shell(BACKUP_CONFIG_DB_CMD)
+
+    duthost.shell("sonic-clear fdb all")
+    generate_dut_config_files(duthost, minigraph_facts, vnet_test_params, vnet_config)
+    apply_dut_config_files(duthost, vnet_test_params)
+
+    # In this case yield is used only to separate this fixture into setup and teardown portions
+    yield
+
+    if vnet_test_params[CLEANUP_KEY]:
+        logger.info("Restoring config_db.json")
+        duthost.shell(RESTORE_CONFIG_DB_CMD)
+        duthost.shell(DELETE_BACKUP_CONFIG_DB_CMD)
+
+        duthost.shell(CONFIG_RELOAD_CMD)
+    else:
+        logger.info("Skipping cleanup")
+
+
+def bgp_connected(duthost):
+    """
+    Checks if BGP connections are up
+
+    BGP connections are "up" once they have received all prefixes (6400) from all neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match entire line containing BGP neighbor information such as:
+    # 10.0.0.1        4      64802    6419   10273        0    0    0 00:09:36         6400   ARISTA01T1
+    bgp_neighbor_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}.*")
+
+    bgp_neighbors = []
+
+    for line in bgp_summary:
+        matched = bgp_neighbor_regex.match(line)
+
+        if matched:
+            bgp_neighbors.append(str(matched.group(0)))
+
+    if not bgp_neighbors:
+        return False
+
+    for neighbor in bgp_neighbors:
+        if neighbor.split()[9] != "6400":
+            return False
+
+    logger.info("BGP sessions up")
+
+    return True
+
+
+def get_leaked_routes(duthost):
+    """
+    Gets all VNET routes and checks that they are not advertised to any BGP neighbors
+
+    Args:
+        duthost: DUT host object
+    """
+
+    vnet_routes = duthost.shell(SHOW_VNET_ROUTES_CMD)["stdout"].split("\n")
+    logger.debug("VNET prefixes: {}".format(vnet_routes))
+
+    vnet_prefixes = []
+
+    for line in vnet_routes:
+        # Ignore header lines and separators
+        # All other lines will contain numbers in the form of an IP address/prefix, which is the information we want to extract
+        if any(char.isdigit() for char in line):
+            vnet_prefixes.append(line.split()[1])
+
+    bgp_summary = duthost.shell(SHOW_BGP_SUMMARY_CMD)["stdout"].split("\n")
+    logger.debug("BGP Summary: {}".format(bgp_summary))
+
+    # Match only IP addresses at the beginning of the line
+    # Only IP addresses of neighbors should be matched by this
+    bgp_neighbor_addr_regex = re.compile(r"^([0-9]{1,3}\.){3}[0-9]{1,3}")
+
+    bgp_neighbors = []
+
+    for line in bgp_summary:
```

Comment:
As these match regex are repeated in this function, can you move it to a common function?

Reply:
The matches are slightly different, this one only matches the IP of the neighbor, the other regex matches the entire line containing neighbor info. Could still pull it out to a common function if you want.

Reply:
Fixed in next push